environment/src/main/java/com/sequenceiq/environment/environment/flow/creation/handler/EnvironmentInitHandler.java

```diff
@@ -52,6 +62,34 @@ public void accept(Event<EnvironmentDto> environmentDtoEvent) {
                 );
     }
 
+    private void initEnvironment(Environment environment) {
+        String environmentCrnForVirtualGroups = getEnvironmentCrnForVirtualGroups(environment);
+        if (!createVirtualGroups(environment, environmentCrnForVirtualGroups)) {
+            // To keep backward compatibility, if somebody passes the group name, then we shall just use it
+            environmentService.setAdminGroupName(environment, environment.getAdminGroupName());
+        }
+        environmentService.assignEnvironmentAdminAndOwnerRole(environment.getCreator(), environmentCrnForVirtualGroups);
+        setLocationAndRegions(environment);
+    }
+
+    private String getEnvironmentCrnForVirtualGroups(Environment environment) {
+        String environmentCrnForVirtualGroups = environment.getResourceCrn();
+        if (Objects.nonNull(environment.getParentEnvironment())) {
+            environmentCrnForVirtualGroups = environment.getParentEnvironment().getResourceCrn();
+        }
+        return environmentCrnForVirtualGroups;
+    }
+
+    private boolean createVirtualGroups(Environment environment, String envCrn) {
+        boolean result = false;
+        if (StringUtils.isEmpty(environment.getAdminGroupName())) {
```

Comment:
isNullorEmpty ?

Reply:
isblank? :D

Reply:
dont worry they all do the same. :D I like the isEmpty named version